cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterPumpDisassociateController.java

```diff
@@ -86,41 +87,18 @@ public WaterPumpDisassociateController(MetricRegistry metrics) {
     public void handle(@NotNull Context ctx) {
         try (Timer.Context ignored = markAndTime(DELETE)) {
             DSLContext dsl = getDslContext(ctx);
-            boolean deleteAccounting = Boolean.parseBoolean(ctx.queryParam(METHOD));
+            boolean deleteAccounting = Boolean.parseBoolean(requiredParam(ctx, METHOD));
             String officeId = ctx.pathParam(OFFICE);
+            String pumpName = ctx.pathParam(NAME);
             String projectName = ctx.pathParam(PROJECT_ID);
             String entityName = ctx.pathParam(WATER_USER);
-            PumpType pumpType = PumpType.valueOf(ctx.queryParam(PUMP_TYPE));
+            PumpType pumpType = PumpType.valueOf(requiredParam(ctx, PUMP_TYPE));
```

Comment:
Use `requiredParamAs` and pass in the `PumpType.class`

Reply:
Replaced with requiredParamAs, added javalin validator